src/structures/paging/mapper/mod.rs

```diff
@@ -374,6 +382,25 @@ pub trait Mapper<S: PageSize> {
     }
 }
 
+/// The result of [`Mapper::clear`], representing either
+/// the unmapped frame or the entry data if the frame is not marked as present.
+#[derive(Debug)]
+#[must_use = "Page table changes must be flushed or ignored if the page is present."]
+pub enum UnmappedFrame<S: PageSize> {
+    /// The frame was present before the [`Mapper::clear`] call
+    Present {
+        /// The physical frame that was unmapped
+        frame: PhysFrame<S>,
+        /// The changed page, to flush the TLB
+        flush: MapperFlush<S>,
+    },
+    /// The frame was not present before the [`Mapper::clear`] call
+    NotPresent {
+        /// The page table entry
+        entry: PageTableEntry,
+    },
+}
```

Comment:
I'm not completely happy with this: The `NotPresent` variant exposes all bits of the cleared page table entry whereas `Present` doesn't. It feels a bit wrong to return *less* data in the case of a successful operation. @phil-opp @josephlr does either one of you have any opinions on this?

Reply:
How about we just add the `PageTableFlags` to the `Present` variant? Then we should have all the information too, just in a more structured way.

Reply:
Add it as a third field or should I just change the `frame` to `PageTableEntity`? Also should that change be reflected in the `unmap` API?

Reply:
Adding `PageTableFlags` to the `Present` variant as a third field and adding `PageTableFlags` to the return type of unmap sounds good to me.